src/main/java/com/example/weather/client/events/WeatherDataCascadeSaveMongoEventListener.java

```diff
@@ -0,0 +1,30 @@
+package com.example.weather.client.events;
+
+import com.example.weather.client.models.entity.WeatherData;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener;
+import org.springframework.data.mongodb.core.mapping.event.BeforeConvertEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class WeatherDataCascadeSaveMongoEventListener extends AbstractMongoEventListener<Object> {
+
+    private final MongoOperations mongoOperations;
+
+    @Override
+    public void onBeforeConvert(final BeforeConvertEvent<Object> event) {
+        if (event.getSource() instanceof WeatherData source) {
+            if (source.getWeathers() != null &&
+                    source.getCoordinates() != null &&
+                    source.getConditions() != null &&
+                    source.getWind() != null) {
```

Comment:
Not to do in this PR, but in some future improvement task create an interceptor on the restTemplate, that assures the relevant data exists. That way you'll be able to omit this safety check.